build-tools/packages/build-cli/src/library/commands/generateEntrypoints.ts

```diff
@@ -455,10 +459,30 @@ async function generateEntrypoints(
 			sourceFile.insertText(0, `${newFileHeader}export {}\n\n`);
 		}
 
-		fileSavePromises.push(sourceFile.save());
-	}
+		await sourceFile.save();
+	};
 
-	await Promise.all(fileSavePromises);
+	await Promise.all([
+		writeExports("public", namedExportMap.public),
+		writeExports("beta", [...namedExportMap.public, ...namedExportMap.beta]),
+		writeExports("alpha", [
+			...namedExportMap.public,
+			...namedExportMap.beta,
+			...namedExportMap.alpha,
+		]),
+		writeExports("legacy", [
+			...namedExportMap.public,
+			...namedExportMap.legacy,
+			...(supportLegacyAlpha ? [] : namedExportMap.legacyAlpha),
+		]),
+		writeExports("legacyAlpha", [
```

Comment:
i made how we compute our level exports much more explicit. for legacyAlpha  think we should include beta and alpha, so we can combine with regular beta and alpha

Reply:
API council had also talked about moving the beta break cadence to the same as the legacy break cadence, if we did that, beta could also be in normal legacy if we wanted, but not sure we would, as they are somewhat different goals.